### ISSUE TITLE
bot should warn again if time has passed

### ISSUE THREAD

**Issue by bsipocz** [astropy/astropy-bot#63]
Currently the bot warns for stale PRs, then after a month it closes said PRs. However with a rebase the clock can be reset, but then it won't warn again, but close straight away.
I feel it would be more helpful to receive the warning again.

Example is here: https://github.com/astropy/astropy/pull/5195

**Comment by pllim** [comment]
I looked at the logic and not sure how to accomplish this. Currently the logic is (pseudocode follows):
```
if delta_time_between_now_and_last_commit > close_threshold:
    close_pr
elif delta_time_between_now_and_last_commit > warn_threshold:
    warn_pr
```
~~Therefore, the warning will only happen once and only once. Rebase does not alter the timestamp of last commit, it simply replays the commits on top of `master`. Currently the only way for it to warn again instead of simply closing is to add another commit.~~

Here is the data it has to work with: https://api.github.com/repos/astropy/astropy/pulls/5195/commits

**Comment by bsipocz** [comment]
As I see it uses the commit date when setting `last_commit_date`. Rebase updates that the commit date, only leaves the author date untouched.

**Comment by pllim** [comment]
Ah, yes. Then maybe @astrofrog can comment. Could be unfortunate scheduling such that the bot was not checking when `warn_threshold` was exceeded again (but before `close_threshold`). That way, next time bot runs this check, it goes straight to closing the PR?

**Comment by bsipocz** [comment]
Yes, I think so that's the case. However we could easily argue that a PR that was once stale and got stale again is most probably very stale indeed (as after having a careful look, it was the case this time, too).

**Comment by pllim** [comment]
Oh, wait... Looks like it purposely only warn once. This seems to be a conscious design choice at some point. Anyone remembers why?
```
            else:
                print(f'-> Skipping issue {n} (already warned)')
```

**Comment by bsipocz** [comment]
I think we were thinking that it shouldn't warn each week, or each day during the grade period. I can't recall that we have thought about the clock restart (though me might have had).

**Comment by pllim** [comment]
It is not impossible to insert extra logic to not skip warning again if the older warning is too old, but we need to think about the cadence (if that is indeed the correct English word to use).

**Comment by pllim** [comment]
Ah, now that I look at the issues. Is this a duplicate of #38?

**Comment by bsipocz** [comment]
Ah, sure it is. I haven't checked the issues.
